### ISSUE TITLE
Blurred background video instead of black bars?

### ISSUE THREAD

**Issue by Y2K350** [dvlden/uw-pub#23]
**Is your feature request related to a problem? Please describe.**
My Feature is not related to a problem with this extension.

**Describe the solution you'd like**
when not using the extension, 16:9 content (or other videos with an aspect ratio different to your monitors) are shown with black bars on the sides. I was hoping for an option where instead of black bars, a blurred video plays on the sides that is loosely based on the content displayed on the screen (screenshot examples listed below). This would help OLED 21:9 monitor owners prevent burn in on their displays when viewing 16:9 content.

**Describe alternatives you've considered**
Obviously you could use ultrawideo to crop the video to 21:9, however sometimes you do lost important information in certain media content (namely videos that use rendered in subtitles, stats, or just generally rely on you seeing the entire screen.

**Additional context**
I understand this is not necessarily in scope for the project, but I believe it would basically cement the app as a must have have app for all 21:9 (or other non-standard aspect ratio) monitor owners and I hope you may at least consider it.

Below are two example of what the video would look like before and hopefully after. (the images represent someone using Photoshop to correct the issue, but the idea would be the same for video playback).

![Screenshot from 2024-04-21 14-45-25](https://github.com/dvlden/uw-pub/assets/140640993/01f622e3-ff04-4fee-ba18-e95f5037dd1a)

![Screenshot from 2024-04-21 14-46-21](https://github.com/dvlden/uw-pub/assets/140640993/b96dd46c-9ae7-49dc-9b91-65ad43da081a)

**Comment by Y2K350** [comment]
Edited to add examples.

**Comment by nnx0r** [comment]
Hello and thank you for your request!

I never thought someone would request such feature, but I made a prototype like 2 years ago.
Tried doing it with double videos, but it was bad, because it loads two sources and drains the internet bandwidth, not to mention the battery.

However, after a couple of trial and errors, I made it fast and performant, using canvas. Find the sample attached below this paragraph...

https://github.com/dvlden/uw-pub/assets/3460368/1b68778a-56d3-400e-815a-6b37e301926c

Honestly, I did not have any plans on implementing this into the extension, but hearing people that they'd rather prefer some black-bars, over content loss, sounds like this solution may not just be fun, but pleasing to the eye. 

It's too early to say anything, but at this stage, I think it might be very difficult to implement, as UltraWideo aims to work on all platforms and embedded videos... So I'll keep this open and see if any other users come around and vote for this feature.

**Comment by toindev** [comment]
> This would help OLED 21:9 monitor owners prevent burn in on their displays when viewing 16:9 content.

Aren't black pixels just turned off on OLED displays?

**Comment by nnx0r** [comment]
@toindev Fair point. I believe I missed that information along the way, while reading OP's request. 

Nonetheless, I think it'd be cool to implement this at some point. Might make vertical videos more enjoyable to watch.

**Comment by Y2K350** [comment]
@toindev Yes black bars are turned off on Oleds, but thats exactly the problem. As the pixels in the center of the screen are on basically all the time, but the edges where black bars appear are off for a decent amount of time. This causes uneven wear in the monitor and makes a fine line where the black bars are due to the central pixels wearing out faster than the edges where the bars would be.

**Comment by toindev** [comment]
Hooo, it makes sense now, thanks

**Comment by nnx0r** [comment]
This feature development is in progress, but will be available only for **Pro** users. 
It will work for both horizontal and vertical videos and can be toggled on/off through interface (popup) or keyboard shortcut.

It will be platform independent, just like entire UltraWideo logic.

**Comment by nnx0r** [comment]
Published this feature for Pro users. 
I will close it now as resolved, but be patient as Chrome and Firefox still have to release the version.

**Comment by Y2K350** [comment]
> Published this feature for Pro users. I will close it now as resolved, but be patient as Chrome and Firefox still have to release the version.

any chance I could get a free trial for suggesting the idea lmao.

**Comment by nnx0r** [comment]
Quote from my first reply:
> I never thought someone would request such feature, but I made a prototype like 2 years ago.

However, yes, sure. I can get you a free trial when Chrome and Firefox approves the published version. It's still pending.

**Comment by suklaakados** [comment]
@dvlden Hello, im not a pro user but now after latest update i have this blurred background. How can i turn it off? i want it to be black like it used to be

**Comment by nnx0r** [comment]
@suklaakados I will look into it today. You shouldn't be seeing this and its supposed to be black for non-pro users.

**EDIT:** Yikes, I did forget to check if user is upgraded in one place that actually made ambient light active for all users. Thank you for letting me know about this, I wouldn've never realise and it may have caused some slowdowns to users with low system specs.

Silent fix is on the way, awaiting Chrome Web Store to approve it.

**Comment by suklaakados** [comment]
@dvlden thank you for responding quickly and fixing the issue! Its black now, like it was. Also, while we are here, i wanted to tell you that blurred background was shimmering a lot when i had the feature. I dont know though whether that was an issue with my gpu or extension itself.
